assets/src/components/v2/elevator/elevator_closures.tsx

```diff
@@ -19,44 +71,134 @@ const InStationSummary = ({ alerts }: InStationSummaryProps) => {
           <NormalService height={72} width={72} fill="#145A06" />
         </span>
       </div>
-      <hr />
+      <hr className="thick" />
     </>
   );
 };
 
-interface OutsideAlertListProps {
-  alerts: string[];
+interface OutsideClosureListProps extends WrappedComponentProps {
+  stations: StationWithClosure[];
+  lastUpdate: number | null;
 }
 
-const OutsideAlertList = (_props: OutsideAlertListProps) => {
+const OutsideClosureList = ({
+  stations,
+  lastUpdate,
+  onFinish,
+}: OutsideClosureListProps) => {
+  const ref = useRef<HTMLDivElement>(null);
+
+  // Each value represents the pageIndex the row is visible on
+  const [rowPageIndexes, setRowPageIndexes] = useState<{
+    [key: number]: number;
+  }>({});
+
+  const numPages = Object.keys(rowPageIndexes).length;
+  const pageIndex = useClientPaging({ numPages, onFinish, lastUpdate });
+
+  const numOffsetRows = Object.keys(rowPageIndexes).reduce((acc, key) => {
+    if (parseInt(key) === pageIndex) {
+      return acc;
+    } else {
+      return acc + rowPageIndexes[key];
+    }
+  }, 0);
+
+  useLayoutEffect(() => {
+    if (!ref.current) return;
+
+    const offsets = Array.from(ref.current.children).map((closure) => {
+      return (closure as HTMLDivElement).offsetLeft;
+    });
+
+    const rowPageIndexes: number[] = [];
```

Comment:
I guess I shouldn't be surprised that `number[]` is type-compatible with `{[key: number]: number}`, but it is a little bit confusing on first read. I think there's some opportunity for cleanup here, starting by replacing L92 with something like

```ts
const [rowCounts, setRowCounts] = useState<number[]>([]);
```